tm1638/tm1638.go

```diff
@@ -0,0 +1,137 @@
+/*
+TM1638 is an LED Controller
+*/
+package tm1638
+
+import (
+	"machine"
+	"time"
+)
+
+// Device wraps the pins of the TM1638.
+type Device struct {
+	/* STB - When this pin is "LO" chip accepting transmission */
+	strobe machine.Pin
+	/* CLK - DIO pin reads data at the rising edge and outputs at the falling edge */
+	clock machine.Pin
+	/* DIO - This pin outputs/inputs serial data */
+	data machine.Pin
+}
+
+const (
+	addressAutoIncrement = 0x40
+	fixedAddress         = 0x44
+	baseAddress          = 0xC0
+	maxAddress           = 0x0F
+	readKeys             = 0x42
+	brightness           = 0x88
+	maxBrightness        = 0x07
+	zeroBrightness       = 0x80
+)
+
+// Create new TM1638 device
+func New(strobe machine.Pin, clock machine.Pin, data machine.Pin) Device {
+	strobe.Configure(machine.PinConfig{Mode: machine.PinOutput})
+	clock.Configure(machine.PinConfig{Mode: machine.PinOutput})
+	return Device{strobe: strobe, clock: clock, data: data}
+}
+
+// Configure TM1638
+func (d *Device) Configure() {
+	d.Clear()
+	d.Brightness(maxBrightness)
+}
+
+// Clear display memory
+func (d *Device) Clear() {
+	d.sendCommand(addressAutoIncrement)
+	d.data.Configure(machine.PinConfig{Mode: machine.PinOutput})
+	d.strobe.Low()
+	d.transmissionDelay()
+	d.write(baseAddress)
+	for i := 0; i < 16; i++ {
+		d.write(0)
+	}
+	d.strobe.High()
+}
+
+// Set display brightness
+func (d *Device) Brightness(value uint8) {
+	if value == 0 {
+		d.sendCommand(zeroBrightness)
+	} else {
+		d.sendCommand(brightness | (value & maxBrightness))
+	}
+}
+
+// Write one display memory element
+func (d *Device) Write(offset uint8, data uint8) {
+	d.sendCommand(fixedAddress)
+	d.strobe.Low()
+	d.transmissionDelay()
+	d.data.Configure(machine.PinConfig{Mode: machine.PinOutput})
+	d.write(baseAddress | (offset & maxAddress))
+	d.write(data)
+	d.strobe.High()
+}
+
+// Write array to display memory
+func (d *Device) WriteArray(offset uint8, data []uint8) {
```

Comment:
What do you think of the `WriteAt` API? I think it would fit here really well.

Reply:
I searching for that topic. I prefer to use API (interface).

Reply:
`int64` has very little overhead on a Cortex-M chip: it takes up just two registers. In fact, `uint32` is equal to (or better!) than `uint8` on Cortex-M chips because the registers are 32-bits in size.
You don't need to worry about the `int64` in this case.